tardis/io/logger/logger.py

```diff
@@ -1,165 +1,268 @@
 import logging
-import sys
+import os
+import re
+from dataclasses import dataclass, field
+import panel as pn
+from IPython.display import display
+from functools import lru_cache
 
-from tardis.io.logger.colored_logger import ColoredFormatter
+PYTHON_WARNINGS_LOGGER = logging.getLogger("py.warnings")
 
-logging.captureWarnings(True)
-logger = logging.getLogger("tardis")
+@lru_cache(maxsize=1)
+def get_environment():
+    """Determine the execution environment"""
+    try:
+        import IPython
+        ipython = IPython.get_ipython()
 
-console_handler = logging.StreamHandler(sys.stdout)
-console_formatter = ColoredFormatter()
-console_handler.setFormatter(console_formatter)
+        if ipython is None:
+            return 'standard'
 
-logger.addHandler(console_handler)
-logging.getLogger("py.warnings").addHandler(console_handler)
+        # Check for VSCode specific environment variables
+        if any(x for x in ('VSCODE_PID', 'VSCODE') if x in os.environ):
+            return 'vscode'
 
-LOGGING_LEVELS = {
-    "NOTSET": logging.NOTSET,
-    "DEBUG": logging.DEBUG,
-    "INFO": logging.INFO,
-    "WARNING": logging.WARNING,
-    "ERROR": logging.ERROR,
-    "CRITICAL": logging.CRITICAL,
+        # Check if running in Jupyter notebook
+        if 'IPKernelApp' in ipython.config:
+            return 'jupyter'
+
+        return 'standard'
+    except:
+        return 'standard'
+
+def create_output_widget(height=300):
+    return pn.pane.HTML(
+        "",
+        height=height,
+        styles={
+            'overflow-y': 'auto',
+            'overflow-x': 'auto',
+            'border': '1px solid #ddd',
+            'width': '100%',
+            'font-family': 'monospace',
+            'padding': '8px',
+            'background-color': 'white'
+        }
+    )
+
+ENVIRONMENT = get_environment()
+LOG_OUTPUTS = {
+    "WARNING/ERROR": create_output_widget(),
+    "INFO": create_output_widget(),
+    "DEBUG": create_output_widget(),
+    "ALL": create_output_widget(),
 }
-DEFAULT_LOG_LEVEL = "INFO"
-DEFAULT_SPECIFIC_STATE = False
+TAB_ORDER = ["ALL", "WARNING/ERROR", "INFO", "DEBUG"]
+LOGGER_WIDGET = pn.Tabs(
+    *[(title, LOG_OUTPUTS[title]) for title in TAB_ORDER],
+    height=350,
+    sizing_mode='stretch_width'
+)
 
+if ENVIRONMENT == 'vscode':
+    pn.extension()
+    display(LOGGER_WIDGET)
+elif os.environ.get("PANEL_DOCS_BUILD"):
```

Comment:
I do not like environment variables being used as globals like this. Perhaps it should set a global instead?